internal/pkg/skuba/deployments/ssh/kubelet.go

```diff
@@ -62,84 +55,6 @@ func kubeletUploadRootCert(t *Target, data interface{}) error {
 	return nil
 }
 
-func kubeletCreateAndUploadServerCert(t *Target, data interface{}) error {
-	// Read kubelet root ca certificate and key
-	caCert, caKey, err := pkiutil.TryLoadCertAndKeyFromDisk(skuba.PkiDir(), kubernetes.KubeletCACertAndKeyBaseName)
-	if err != nil {
-		return errors.Wrap(err, "failure loading kubelet CA certificate authority")
-	}
-
-	host := t.target.Nodename
-	altNames := certutil.AltNames{}
-	if ip := net.ParseIP(host); ip != nil {
-		altNames.IPs = append(altNames.IPs, ip)
-	} else {
-		altNames.DNSNames = append(altNames.DNSNames, host)
-	}
-
-	// Create AltNames with defaults DNSNames/IPs
-	stdout, _, err := t.silentSsh("hostname", "-I")
-	if err != nil {
-		return err
-	}
-	for _, addr := range strings.Split(stdout, " ") {
-		if ip := net.ParseIP(addr); ip != nil {
-			altNames.IPs = append(altNames.IPs, ip)
-		}
-	}
-
-	alternateIPs := []net.IP{net.IPv4(127, 0, 0, 1), net.IPv6loopback}
-	alternateDNS := []string{"localhost"}
-	if ip := net.ParseIP(t.target.Target); ip != nil {
-		alternateIPs = append(alternateIPs, ip)
-	} else {
-		alternateDNS = append(alternateDNS, t.target.Target)
-	}
-
-	altNames.IPs = append(altNames.IPs, alternateIPs...)
-	altNames.DNSNames = append(altNames.DNSNames, alternateDNS...)
-
-	certCfg := certutil.Config{
-		CommonName: host,
-		AltNames:   altNames,
-		Usages:     []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
-	}
-	cfg := &pkiutil.CertConfig{
-		Config: certCfg,
-	}
-	cert, key, err := pkiutil.NewCertAndKey(caCert, caKey, cfg)
-	if err != nil {
-		return errors.Wrap(err, "couldn't generate kubelet server certificate")
-	}
-
-	// Save kubelet server certificate and key to local temporarily
-	if err := pkiutil.WriteCertAndKey(skuba.PkiDir(), host, cert, key); err != nil {
-		return errors.Wrapf(err, "failure while saving kubelet server %s certificate and key", host)
-	}
-
-	// Upload server certificate and key
-	certPath, keyPath := pkiutil.PathsForCertAndKey(skuba.PkiDir(), host)
-	if err := t.target.UploadFile(certPath, filepath.Join(kubernetes.KubeletCertAndKeyDir, kubernetes.KubeletServerCertName)); err != nil {
-		return err
-	}
-	if err := t.target.UploadFile(keyPath, filepath.Join(kubernetes.KubeletCertAndKeyDir, kubernetes.KubeletServerKeyName)); err != nil {
-		return err
-	}
-	if _, _, err := t.silentSsh("chmod", "0400", filepath.Join(kubernetes.KubeletCertAndKeyDir, kubernetes.KubeletServerKeyName)); err != nil {
-		return err
-	}
-
-	// Remove local temporarily kubelet server certificate and key
-	if err := os.Remove(certPath); err != nil {
-		return err
-	}
-	if err := os.Remove(keyPath); err != nil {
-		return err
-	}
-
-	return nil
-}
-
 func kubeletConfigure(t *Target, data interface{}) error {
```

Comment:
Removing `kubelet server cert creation and upload` is because we enable `server rotation` so it's expected everything will be setup automatically?

Reply:
yes, I intended to add kucero at previous version `1.17.4` is because I want to install kucero first and enable kubelet `serverTLSBootstrap: true` when cluster upgrade.